src/XPurchase/VirtoCommerce.XPurchase/CartProduct.cs

```diff
@@ -62,16 +62,20 @@ public virtual void ApplyPrices(IEnumerable<PricingModule.Core.Model.Price> pric
             AllPrices.Clear();
             Price = null;
 
+            // Return correct currency from price
+            Currency GetCurrencyFromPrice(PricingModule.Core.Model.Price price) => new Currency(new CoreModule.Core.Common.Language(currency.CultureName), price.Currency);
```

Comment:
You can't create the currency in place. all currencies must be loaded from the store because each currency might have the customer settings like formatting template or currency symbol.